applications/rag/frontend/main.tf

```diff
@@ -30,6 +34,46 @@ locals {
     instance_connection_name = format("%s:%s:%s", var.project_id, var.region, "pgvector-instance")
 }
 
+# IAP Section: Enabled the IAP service
+resource "google_project_service" "project_service" {
+  project = var.project_id
+  service = "iap.googleapis.com"
+
+  disable_dependent_services = false
+  disable_on_destroy         = false
+}
+
+# IAP Section: Creates the OAuth client used in IAP
+resource "google_iap_client" "iap_oauth_client" {
```

Comment:
@blackzlq @bjornsen do we need separate clients for frontend/jupyter? can they share one to make the setup simple?

Reply:
Discussed offline, the user case for frontend and jupyter may different, so we should do that.